File: python/analyze_orders.py
from pathlib import Path

import pandas as pd


data_file = Path(__file__).parent.parent / "data" / "orders_raw.csv"

df = pd.read_csv(data_file)

print(df.head())
print(df.shape)
print(df.columns)
df.info()

df["OrderDate"] = pd.to_datetime(df["OrderDate"])

df["Revenue"] = df["Quantity"] * df["UnitPrice"]

df["Late"] = (
    df["ShippingDays"] > df["ExpectedShippingDays"]
)

print(df.groupby("PartCategory")["Revenue"].sum())

File: r/analyze_orders.R
setwd("C:/Users/Jeremy/source/repos/nosirrah01/medical-parts-data-analysis")
getwd()
file.exists("data/orders_raw.csv")
orders <- read.csv("data/orders_raw.csv")

head(orders)
str(orders)
summary(orders)

orders$Revenue <- orders$Quantity * orders$UnitPrice

orders$ShippingStatus <- ifelse(
  orders$ShippingDays > orders$ExpectedShippingDays,
  "Late",
  "On Time"
)

head(orders)

aggregate(
  Revenue ~ PartCategory,
  data = orders,
  FUN = sum
)

install.packages("tidyverse")

library(tidyverse)

orders <- read_csv("data/orders_raw.csv")

orders <- orders %>%
  mutate(
    Revenue = Quantity * UnitPrice,
    ShippingStatus = if_else(
      ShippingDays > ExpectedShippingDays,
      "Late",
      "On Time"
    )
  )

orders %>%
  group_by(PartCategory) %>%
  summarise(
    TotalOrders = n(),
    TotalRevenue = sum(Revenue),
    AvgShippingDays = mean(ShippingDays)
  )